lib/fabrik/Fabrik.js

```diff
@@ -31,12 +32,27 @@ class Fabrik {
       .then(managerConstructor => managerConstructor.load(plan));
   }
 
-  static createInstance(instance_id, service_id, plan_id) {
+  static createInstance(instance_id, service_id, plan_id, context) {
     const plan = catalog.getPlan(plan_id);
     assert.strictEqual(service_id, plan.service.id);
     return this
       .createManager(plan)
-      .then(manager => manager.createInstance(instance_id));
+      .then(manager => {
+        const instance = manager.createInstance(instance_id);
+        return Promise
+          .try(() => {
```

Comment:
can covert all of this try section to the below oneline
```
return Promise.try(() => context ? context : instance.platformContext)
....
```

Reply:
sure. would do that.